### PR TITLE
CTCP-2991: Items transport equipment page.

### PR DESCRIPTION
**Pull request by justega247**

### CODE REVIEW

File: app/services/TransportEquipmentService.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import models.{SelectableList, TransportEquipment, UserAnswers}
+import pages.sections.external.TransportEquipmentsSection
+
+import javax.inject.Inject
+
+class TransportEquipmentService @Inject() () {
+
+  def getTransportEquipments(userAnswers: UserAnswers): SelectableList[TransportEquipment] = {
+    val equipmentCount = userAnswers.get(TransportEquipmentsSection).get.value.length
```
Comment (jackheslop96) [inline app/services/TransportEquipmentService.scala:27]: Avoid using .get if you can.

`userAnswers.get(TransportEquipmentsSection).map(_.value.length).getOrElse(0)` might be better, or consider using https://github.com/hmrc/ctc-departure-documents-frontend/blob/22720ea296d45f41c42286de82181c13a1930045/app/models/UserAnswers.scala#L38

File: app/services/TransportEquipmentService.scala [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    val transportEquipments = (1 to equipmentCount).map(
+      x => TransportEquipment(x)
+    )
```
Comment (jackheslop96) [inline app/services/TransportEquipmentService.scala:30]: Or you could just do
`val transportEquipments = userAnswers.get(TransportEquipmentsSection).map(_.value).getOrElse(Nil).zipWithIndex.map(_._2 + 1)`

**Review by jackheslop96 — APPROVED** [review]

**Review by BledarFeratiEY — APPROVED** [review]

File: conf/messages.en [hunk 3]
```diff
@@ -432,4 +434,12 @@ item.supplyChainActors.addAnotherSupplyChainActor.error.required = Select yes if
...
 item.supplyChainActors.index.removeSupplyChainActor.title = Are you sure you want to remove this supply chain actor?
 item.supplyChainActors.index.removeSupplyChainActor.heading = Are you sure you want to remove this supply chain actor?
-item.supplyChainActors.index.removeSupplyChainActor.error.required = Select yes if you want to remove this supply chain actor
\ No newline at end of file
+item.supplyChainActors.index.removeSupplyChainActor.error.required = Select yes if you want to remove this supply chain actor
+
+item.transportEquipment.title = Which transport equipment are you using for this item?
+item.transportEquipment.heading = Which transport equipment are you using for this item?
+item.transportEquipment.paragraph = Transport equipment can be anything used to transport goods. For example, a swap body, articulated lorry or trailer.
+item.transportEquipment.placeholder = Select a transportEquipment
```
Comment (jackheslop96) [inline conf/messages.en:442]: Scaffolding issue I'm guessing? Should be `Select a transport equipment`

**Review by jackheslop96 — APPROVED** [review]

File: app/services/TransportEquipmentService.scala [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      }
+      .toSeq
+    SelectableList(transportEquipments)
+  }
+
+  def getTransportEquipment(userAnswers: UserAnswers, itemIndex: Index): Option[TransportEquipment] =
+    for {
+      uuid <- userAnswers.get(TransportEquipmentPage(itemIndex))
+      transportEquipments =
+        userAnswers.get(TransportEquipmentsSection).map(_.validateAsAListOf[TransportEquipment]).getOrElse(Nil)
```
Comment (jackheslop96) [inline app/services/TransportEquipmentService.scala:44]: We have a slight discrepancy here. I think what we should do is add 1 in the reads equipmentReads method itself. What we could also have here is a private def that returns a Seq[TransportEquipment] as this is common to both methods in this class.

**Review by jackheslop96 — DISMISSED** [review]

**Review by jackheslop96 — APPROVED** [review]

**Pull request merged**